pkg/v3/postprocessors/retry.go

```diff
@@ -30,19 +30,23 @@ func (p *retryPostProcessor) PostProcess(_ context.Context, results []ocr2keeper
 }
 
 func (p *retryPostProcessor) attemptRetry(res ocr2keepers.CheckResult) error {
-	err := p.retryer.Retry(res)
-	if err == nil {
-		return nil
+	var err error
+
+	if p.retryer != nil {
+		err = p.retryer.Retry(res)
+		if err == nil {
+			return nil
+		}
 	}
 
-	if errors.Is(err, tickers.ErrSendDurationExceeded) {
-		if err := p.recoverer.Retry(res); err != nil {
+	if err == nil || errors.Is(err, tickers.ErrSendDurationExceeded) {
```

Comment:
I find this logic pretty tricky to follow; can we have full test coverage for this function so that we have all possible flows covered? Or could we rewrite this so that the error cases are more closely tied to the calls that return those errors?